src/Plugin/Validators/EmptyCell.php

```diff
@@ -78,12 +104,137 @@ public function validateRow($row_values) {
     }
     else {
       $validator_status = [
-        'case' => 'No empty values found in required column(s)',
+        'case' => self::$mapping['case-valid']['dev-case'],
         'valid' => TRUE,
         'failedItems' => [],
       ];
     }
+
     return $validator_status;
   }
 
+  /**
+   * Processes failed validation from EmptyCell into a render array.
+   *
+   * @param array $failures
+   *   An associative array that stores the validation failures by the
+   *   EmptyCell validator. It is keyed by the line number of the input
+   *   file where validation failed, and the value is an associative array
+   *   returned by the validator. Here is the overall structure of $failures:
+   *   - [LINE NUMBER]:
+   *     - 'case': a developer-focused string describing the case checked.
+   *     - 'valid': FALSE to indicate that validation failed.
+   *     - 'failedItems': an array of items that failed:
+   *       - 'empty_indices': A list of column indices in the line which were
+   *         checked and found to be empty.
+   * @param array $headers
+   *   Header array that list the headers or columns in a data file. This is the
+   *   headers property defined by an importer instance.
+   * @param array $tokens
+   *   [OPTIONAL] An array of values to use for token replacement.
+   *   @see ProjectGenusMatch::$mapping
+   *   The following tokens can be specfied as keys, with value as the
+   *   replacement value for the token. These apply to all failure cases.
+   *   - 'project': the word to use when referring to the project.
+   *   - 'contact-admin': the phrase to use when the user needs a privileged
+   *     administrator to fix the problem.
+   *   The following token keys will substitute the entire existing case message
+   *   to the user with the value of that token.
+   *   - 'case-no-project': the message when a project does not exist.
+   *   - 'case-no-paired-genus': the message when a project has no genus set
+   *     to it.
+   *   - 'case-project-genus-mismatch': the message when the genus selected
+   *     by the user is not configured to the selected project.
+   *
+   * @return array
+   *   A render array of type unordered list which is used to display feedback
+   *   to the user about the case(s) that failed and the failed items from the
+   *   input file. This unordered list will include a table that lists the row
+   *   and column combinations with empty cells. It has the following headers:
+   *   - 'Line Number'
+   *   - 'Column(s) with empty value'
+   *
+   * @throws \Exception
+   *   - If the validation_result parameter was not formatted properly.
+   *   - If the case string returned by the validator implied validation passed.
+   *   - If the case string returned by the validator is not recognized.
+   */
+  public static function processListWithDescribedTable(array $failures, array $headers, array $tokens = []) {
```

Comment:
Instead of passing the parameter $headers, you should be using $metadata instead. This will allow us to provide additional information to headers in the future if needed without expanding the param list for the method.

See here for the definition of $metadata (you can copy and paste this 😉 ): https://github.com/TripalCultivate/TripalCultivate/blob/3080d00180b0b7fb250ed38c36dd665c2d64c7f6/src/Plugin/Validators/GermplasmNameExists.php#L303-L312

```suggestion
  public static function processListWithDescribedTable(array $failures, array $tokens = [], array $metadata) {
```